mmds/src/lib.rs

```diff
@@ -22,6 +22,26 @@ lazy_static! {
     pub static ref MMDS: Arc<Mutex<Mmds>> = Arc::new(Mutex::new(Mmds::default()));
 }
 
+/// Patch provided JSON document (given as `serde_json::Value`) in place with JSON Merge Patch
+/// (RFC 7396).
+pub fn json_patch(target: &mut Value, patch: &Value) {
+    if patch.is_object() {
+        if !target.is_object() {
+            *target = Value::Object(Map::new());
+        }
+        let doc = target.as_object_mut().unwrap();
+        for (key, value) in patch.as_object().unwrap() {
+            if value.is_null() {
+                doc.remove(key.as_str());
+            } else {
+                json_patch(doc.entry(key.as_str()).or_insert(Value::Null), value);
```

Comment:
When can a key in the dictionary be something else other than a string?

Reply:
It seems like serde_json always uses Strings as key values, so, it's always a string. That being said, key is already a String, here it's being converted to a &str.

Reply:
Yes, that is true. I was a bit confused because I didn't understand why you need the `or_inset(Value::Null)`. Can you add a comment to explain that part?

Reply:
That creates the entry in the map if it's not occupied and returns a mutable reference to it, so it can recursively apply json_patch to it. https://docs.serde.rs/serde_json/map/enum.Entry.html#method.or_insert

I'll add a couple of comments to make it clear what's going on.